src/__tests__/profilerExportImport-test.js

```diff
@@ -0,0 +1,133 @@
+
+    const containerA = document.createElement('div');
+    const containerB = document.createElement('div');
+
+    act(() => {
+      ReactDOM.render(<Parent key="A" count={3} />, containerA);
+      ReactDOM.render(<Parent key="B" count={2} />, containerB);
+    });
+    expect(store).toMatchSnapshot('1: mount');
+
+    act(() => {
+      store.startProfiling();
+    });
+    expect(store).toMatchSnapshot('2: profiling started');
+
+    act(() => {
+      ReactDOM.render(<Parent key="A" count={4} />, containerA);
+      ReactDOM.render(<Parent key="B" count={1} />, containerB);
+    });
+    expect(store).toMatchSnapshot('3: update');
+
+    act(() => ReactDOM.unmountComponentAtNode(containerB));
+    expect(store).toMatchSnapshot('4: unmount B');
+
+    act(() => ReactDOM.unmountComponentAtNode(containerA));
+    expect(store).toMatchSnapshot('5: unmount A');
+
+    act(() => {
+      store.stopProfiling();
+    });
+    expect(store).toMatchSnapshot('6: profiling stopped');
+
+    const profilingSummaryForExportRoot1 = prepareExportedProfilingSummaryFromFrontend(
+      {
+        profilingOperationsByRootID: store.profilingOperations,
+        profilingSnapshotsByRootID: store.profilingSnapshot,
+        rendererID: 1,
+        rootID: 1,
+      }
+    );
+    const profilingSummaryForExportRoot2 = prepareExportedProfilingSummaryFromFrontend(
+      {
+        profilingOperationsByRootID: store.profilingOperations,
+        profilingSnapshotsByRootID: store.profilingSnapshot,
+        rendererID: 1,
+        rootID: 1,
+      }
+    );
+
+    let resolve;
+    let exportedFilesPromise = new Promise(r => {
+      resolve = r;
+    });
+    const exportedFiles = [];
+    bridge.addListener('exportFile', ({ contents, filename }) => {
+      exportedFiles.push({ contents, filename });
+      if (exportedFiles.length === 2) {
+        resolve();
+      }
+    });
+
+    act(() => {
+      bridge.send('exportProfilingSummary', profilingSummaryForExportRoot1);
+    });
+
+    act(() => {
+      bridge.send('exportProfilingSummary', profilingSummaryForExportRoot2);
+    });
+    expect(store).toMatchSnapshot('7: export finished');
+
+    await exportedFilesPromise;
+
+    expect(exportedFiles).toHaveLength(2);
+    expect(exportedFiles[0].filename).toBe('profile-data.json');
+    expect(exportedFiles[1].filename).toBe('profile-data.json');
+
+    // Create new store, bridge, agent.
+    reinit();
+
+    // Import into the new store.
+    store.importedProfilingData = parseExportedProfilingDataString(
+      exportedFiles[0].contents
+    );
+    expect(store).toMatchSnapshot('8: import for root 1 finished');
+    // TODO: Verify that the profiling data matches the data before export.
+
+    store.importedProfilingData = parseExportedProfilingDataString(
+      exportedFiles[1].contents
+    );
+    expect(store).toMatchSnapshot('8: import for root 2 finished');
+    // TODO: Verify that the profiling data matches the data before export.
+  });
```

Comment:
I don't think the `Store` snapshots are adding any value here. We should be snapshotting the profiling data, right?

Reply:
> I don't think the `Store` snapshots are adding any value here. We should be snapshotting the profiling data, right?

Yes, you're right. Copy-pasted, didn't spend enough time on the snapshots part of the test.

Reply:
Profiling data is flaky so cannot be snapshotted. :( TODO

Reply:
Nah :) It's just that timing values vary slightly between runs. This isn't that surprising given that we're using high-res timestamps.

Don't worry about this though. I added tests over the weekend (PR #258) just need to get a new React canary built after https://github.com/facebook/react/pull/15567 is merged.

Reply:
This is sort of what I meant:
https://github.com/bvaughn/react-devtools-experimental/blob/acdc9bae674d27fb107a8ed99680148f07cfd55d/src/__tests__/profiling-test.js#L17-L55

This code takes real profiling data, exports+imports it through our bridge (admittedly this is a little manual at the moment), then tests verify that the before and after state match.

Reply:
Looks good. 👍